### PR TITLE
feat: structured handoff contracts with MCP submit tools

### LINKED ISSUE #80: Git issues - comments not loaded
### Agentmux version
N/A

### Provider(s)
- claude
- codex
- gemini
- opencode

### OS / shell
N/A

### What happened?
When using the GitHub integration, only the issue title and body (description) are loaded — comments are not fetched. In `src/agentmux/integrations/github.py` (lines 32-35):

```python
result = subprocess.run(
    ["gh", "issue", "view", issue_ref, "--json", "title,body"],
    ...
)
```

Only `title` and `body` are requested. Comments are completely ignored, meaning the agent does not get the "full picture" of the issue discussion.

### What did you expect?
The GitHub issue bootstrap should load all comments so that agents have complete context when working on a feature.

### Steps to reproduce
1. Run `agentmux issue <number-or-url>` on an issue that has comments
2. Check the generated feature directory
3. Observe that comments are missing from the bootstrapped context

### Relevant logs or files
- `src/agentmux/integrations/github.py` — `gh issue view` only requests `title,body` fields

### LINKED ISSUE #102: Agents hand offs need structure
Currently there is not structure for agents handoff documents and other artifacts.

We need a strict definiton of what should be described and in what format like interface definiton kind of. 

Could mcp tooling help to better determinstic behaviour?

### LINKED ISSUE #110: Docs: Streamline phase/artifacts lists across documentation files
## Summary

Several documentation files in `docs/` contain repeated lists of workflow phases and artifacts. The same phase structure (01_product_management, 02_planning, etc.) and artifact listings appear across multiple files with minor variations, leading to maintenance overhead and inconsistency risk.

## What problem does this solve?

When updating the workflow (e.g., adding a new phase or changing artifact names), multiple doc files must be manually synchronized. This creates:
1. **Copy-paste maintenance burden** — same lists appear in file-protocol.md, getting-started.md, prompts.md, etc.
2. **Inconsistency risk** — slight variations between files can confuse users
3. **Poor discoverability** — no central reference for the complete phase/artifact catalog

## Area

Other

## Examples from current docs

- `docs/file-protocol.md` — lines 17-50 list all phase directories and artifacts
- `docs/getting-started.md` — contains separate phase overview
- `docs/prompts.md` — has phase-specific prompt path documentation
- `docs/handoff-contracts.md` — mentions handoff artifact structure
- `docs/tmux-layout.md` — references workflow phases

These overlap significantly and should be consolidated.

### PR DESCRIPTION
**Pull request by markuswondrak**
## Summary

Implements structured agent handoff contracts as requested in #102.

### What changed

**YAML migration** — Merged `execution_plan.json` + `plan_meta.json` into a single `execution_plan.yaml` file. All source code, tests, prompts, and docs updated.

**Handoff contracts** (`src/agentmux/workflow/handoff_contracts.py`) — Defines typed contracts for each agent handoff: architecture, execution_plan, subplan, review. Includes field specs, validation logic, and prompt rendering for YAML fallback.

**MCP submit tools** — 4 new tools in the MCP research server:
- `agentmux_submit_architecture()` → validates + writes `architecture.yaml` + `architecture.md`
- `agentmux_submit_execution_plan()` → validates + writes merged `execution_plan.yaml` + `plan.md`
- `agentmux_submit_subplan()` → validates + writes `plan_N.yaml` + `plan_N.md` + `tasks_N.md`
- `agentmux_submit_review()` → validates + writes `review.yaml` + `review.md`

**Handler updates** — Architecting handler now watches `architecture.yaml`; reviewing handler watches `review.yaml`. Both paths converge at the existing state machine transitions.

**Prompt updates** — New shared fragments (`handoff-contract-*.md`) added to architect, planner, and reviewer prompts. Agents get MCP tool instructions + YAML fallback examples.

**Documentation** — New `docs/handoff-contracts.md` + updates to file-protocol, prompts, research-dispatch, monitor, session-resumption docs, and CLAUDE.md.

### Tests
- 57 new tests (34 contract + 23 MCP submit tool tests)
- All 835 tests pass
- Lint and format clean

Closes #102

---

Also closes #80 — GitHub issue bootstrap now loads all comments so agents have complete context.
Also closes #110 - restructured documentation

### CODE REVIEW

File: src/agentmux/integrations/mcp_research_server.py [hunk 1]
```diff
@@ -150,6 +154,292 @@ def agentmux_research_dispatch_web(
     return _dispatch("web", topic, context, questions, scope_hints, feature_dir)
 
 
+# ---------------------------------------------------------------------------
+# Handoff submission helpers
+# ---------------------------------------------------------------------------
```
Comment (markuswondrak) [inline src/agentmux/integrations/mcp_research_server.py]: This functionality does not belong in the research module.

**Review by markuswondrak — COMMENTED** [review]
Not solved

File: src/agentmux/prompts/agents/planner.md [hunk 2]
```diff
@@ -42,29 +43,29 @@ You are the **Execution Planner** (Ausführungsplaner). You receive a completed
...
-    * *Schema:* `{"version": 1, "groups": [{"group_id": "str", "mode": "serial|parallel", "plans": [{"file": "plan_N.md", "name": "Title"}]}]}`
-4.  **`plan_meta.json`**: Risk and doc metadata.
-    * *Schema:* `{"needs_design": bool, "needs_docs": bool, "doc_files": [], "review_strategy": {"severity": "low|medium|high", "focus": []}}`
 
+See Output & Artifacts below for file specs. Write all files to `02_planning/` after explicit user approval.
 
-## Preference memory at phase-end approval
+## Output & Artifacts
+
+- **`02_planning/plan.yaml`** — unified machine-readable plan containing all sub-plans and execution metadata (version 2 schema; the orchestrator materializes derived files automatically).
```
Comment (markuswondrak) [inline src/agentmux/prompts/agents/planner.md]: Implementation Detail - the agents does not need to know what the orchestrator does with that file.

File: src/agentmux/integrations/mcp_server.py [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+        "questions": normalized_questions,
+        "scope_hints": normalized_scope_hints,
+        "research_type": "web",
+    }
+    append_tool_event(_log_path(feature_dir), "research_dispatch_web", payload)
+    return f"Web research on '{normalized_topic}' dispatched."
+
+
+# ---------------------------------------------------------------------------
+# Submission tools
```
[the remark this answers is not in the document]
Comment (markuswondrak) [reply src/agentmux/integrations/mcp_server.py]: In the file, "submission" tools (submit_architecture, submit_plan, submit_review) validate a file written by the agent first (check it exists, parse YAML, run schema validation) and then append an event. They couple file validation with signaling.

The "completion-signal" tools below (submit_done, submit_research_done, submit_pm_done) skip file validation entirely — they just append a structured event to tool_events.jsonl immediately. They are pure status signals with no artifact to validate.

**Comment by markuswondrak** [comment]
@copilot+gpt-5.4 run a review on this PR, focus on: 

- **integrity**: does the implementation match the documentation
- **redudancy**: is code or documentation duplicated or redandant
- **senseable testing**: is the test suite covering senseble test edge cases
